node/service/src/relay_chain_selection.rs

```diff
@@ -419,27 +429,32 @@ where
 		let lag = initial_leaf_number.saturating_sub(subchain_number);
 		self.metrics.note_approval_checking_finality_lag(lag);
 
-		// 3. Constrain according to disputes:
-		let (tx, rx) = oneshot::channel();
-		overseer
-			.send_msg(
-				DisputeCoordinatorMessage::DetermineUndisputedChain {
-					base_number: target_number,
-					block_descriptions: subchain_block_descriptions,
-					tx,
-				},
-				std::any::type_name::<Self>(),
-			)
-			.await;
-		let (subchain_number, subchain_head) = rx
-			.await
-			.map_err(Error::OverseerDisconnected)
-			.map_err(|e| ConsensusError::Other(Box::new(e)))?
-			.unwrap_or_else(|| (subchain_number, subchain_head));
+		let lag = if cfg!(feature = "disputes") {
+			// 3. Constrain according to disputes:
+			let (tx, rx) = oneshot::channel();
+			overseer
+				.send_msg(
+					DisputeCoordinatorMessage::DetermineUndisputedChain {
+						base_number: target_number,
+						block_descriptions: subchain_block_descriptions,
+						tx,
+					},
+					std::any::type_name::<Self>(),
+				)
+				.await;
+			let (subchain_number, _subchain_head) = rx
+				.await
+				.map_err(Error::OverseerDisconnected)
+				.map_err(|e| ConsensusError::Other(Box::new(e)))?
+				.unwrap_or_else(|| (subchain_number, subchain_head));
 
-		// The the total lag accounting for disputes.
-		let lag_disputes = initial_leaf_number.saturating_sub(subchain_number);
-		self.metrics.note_disputes_finality_lag(lag_disputes);
+			// The the total lag accounting for disputes.
+			let lag_disputes = initial_leaf_number.saturating_sub(subchain_number);
+			self.metrics.note_disputes_finality_lag(lag_disputes);
+			lag_disputes
+		} else {
+			lag
+		};
 
 		// 4. Apply the maximum safeguard to the finality lag.
 		if lag > MAX_FINALITY_LAG {
```

Comment:
im reviewing out of context of all this code, but the behavior here before and after is different.

This check before was always checking against `lag`, and with the new code, could check against `lag_disputes` with the feature flag.

Was this a bug before? or a bug now?

Reply:
Bug before, addressed here. Approval and disputes both apply constraints to the finalizable chain and previously the disputes lag wasn't accounted for in the 'max' check.